backend/apps/notificationpref/apis.py

```diff
@@ -0,0 +1,44 @@
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework import status, serializers
+from rest_framework_simplejwt.authentication import JWTAuthentication
+
+from apps.notificationpref.models import UserNotificationPreference
+
+
+class UserNotificationPreferenceView(APIView):
+    permission_classes = [IsAuthenticated]
+    authentication_classes = [JWTAuthentication]
```

Comment:
Use ApiAuthMixin instead of specifying these:

class UserNotificationPreferenceView(APIView, ApiAuthMixin)